streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -714,6 +728,39 @@ public void close(boolean clean,
         taskClosed = true;
     }
 
+    private boolean retrieveCommittedTimestamp(final TopicPartition partition) {
+        final OffsetAndMetadata metadata = consumer.committed(partition);
+
+        if (metadata != null && metadata.metadata().length() != 0) {
+            try {
+                final long committedTimestamp = Long.parseLong(metadata.metadata());
+                partitionGroup.setPartitionTimestamp(partition, committedTimestamp);
+                log.debug("A committed timestamp was detected: setting the partition time of partition {}"
+                          + " to {} in stream task {}", partition, committedTimestamp, this);
+                return true;
+            } catch (final NumberFormatException exc) {
+                log.info("No committed timestamp retrieved because no long integer was found stored in metadata " + exc);
```

Comment:
So we need to log this at INFO level? Seems ERROR might be more appropriate because it actually indicates corrupted metadata? We should also update the error message accordingly:
```
log.error("Could not initialize partition time. Committed metadata is corrupted.", e);
```